### PR TITLE
Documentation/traefik 2

### PR DESCRIPTION
**Pull request by tlex**
Adds a working example with traefik2 and `AUTH_HOST`

#74

### CODE REVIEW

File: examples/docker-compose-auth-host-traefik2.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  traefik-forward-auth:
+    image: thomseddon/traefik-forward-auth:2
+    labels:
+      traefik.enable: 'true'
+      traefik.docker.network: oauth-web
+      traefik.http.middlewares.tfa-verify.forwardAuth.address: http://traefik-forward-auth:4181
+      traefik.http.middlewares.tfa-verify.forwardAuth.trustForwardHeader: 'true'
+      traefik.http.middlewares.tfa-verify.forwardAuth.authResponseHeaders: X-Forwarded-User
+      traefik.http.routers.tfa.entrypoints: web
+      traefik.http.routers.tfa.middlewares: tfa-verify
```
Comment (SuperSandro2000) [inline examples/docker-compose-auth-host-traefik2.yml]: Just as a note mainly to myself: This line is very very important or you will end up in a redirect loop.

File: examples/docker-compose-auth-host-traefik2.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      traefik.http.routers.tfa.rule: "Host(`auth.example.com`)"
+      traefik.http.services.tfaloadbalancer.server.port: '4181'
```
Comment (pew) [inline examples/docker-compose-auth-host-traefik2.yml:45]: there's a `.` missing, changing this made it work for me:

```
traefik.http.services.tfa.loadbalancer.server.port
```

Comment (tlex) [reply examples/docker-compose-auth-host-traefik2.yml]: Thanks for pointing it out. Fixed it.

File: examples/docker-compose-auth-host-traefik2.yml (hunk 1 above)
Comment (wanno-drijfhout) [reply examples/docker-compose-auth-host-traefik2.yml:43]: I'm having the redirect loop as well, so I'm trying to adapt your solution locally. Doesn't seem to work in my circumstances yet.

Would you help me understand why traefik-forward-auth needs to forward auth to itself?

Could you elaborate where the `oauth-web` network is defined?

Would you help me understand if/how multiple networks are being used by traefik-forward-auth? In my case, I only have the public traefik-proxy-network that all my containers share.

Does the `hostname` need to be defined as well for the `forwardAuth.address` to work? (In Docker Swarm, hostnames are randomized.)

File: examples/docker-compose-auth-host-traefik2.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    networks:
+      - traefik
+    environment:
+      PROVIDERS_GOOGLE_CLIENT_ID: 'your-client-id'
+      PROVIDERS_GOOGLE_CLIENT_SECRET: 'your-client-secret'
+      SECRET: 'something-random'
+    command:
+      - ./traefik-forward-auth
+      - --rule.1.action=allow
+      - --rule.1.rule="Path(`/`)"
```
Comment (wanno-drijfhout) [inline examples/docker-compose-auth-host-traefik2.yml:55]: Is this rule necessary? If so, how come?

File: examples/docker-compose-auth-host-traefik2.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      - --log-level=debug
+      - --auth-host=auth.example.com
+      - --cookie-domain=example.com
+      - --insecure-cookie
+
+networks:
+  traefik:
+  traefik-docker:
```
Comment (wanno-drijfhout) [inline examples/docker-compose-auth-host-traefik2.yml:63]: What's the difference between these two networks?

File: examples/docker-compose-auth-host-traefik2.yml (hunk 1 above)
Comment (SuperSandro2000) [reply examples/docker-compose-auth-host-traefik2.yml:43]: Traefik needs this for whatever reason.
Traefik-forward-auth does not care about multiple networks. Traefik and network connection do care.
in swam they still have the predictable name of the service.

**Comment by thomseddon** [comment]
Thank you for these examples, I have added a number of examples over the last few weeks and have updated the README today (8902cf8).

Thank you again for posting these, I know they have been useful to a number of people in getting their config working!

**Pull request closed without merging**